File: internal/indexer/testdata/python/defs_c.py
from models.critical_user_model import CriticalUserModel, create_critical_user_model


class CriticalUserModelCache:
    """Caches CriticalUserModel instances by id."""

    def __init__(self):
        self._entries: dict = {}

    def put(self, id: int, model: CriticalUserModel) -> None:
        self._entries[id] = model

    def get(self, id: int):
        return self._entries.get(id)

    def values(self) -> list:
        return list(self._entries.values())


def transform_critical_user_model(model: CriticalUserModel) -> str:
    """Transforms a CriticalUserModel into a tagged string."""
    return f"[{model.name}]"

File: internal/indexer/testdata/js/defs_c.js
const { CriticalUserModel } = require("./models/criticalUserModel");

class CriticalUserModelCache {
  constructor() {
    this.entries = new Map();
  }

  put(id, model) {
    this.entries.set(id, model);
  }

  get(id) {
    return this.entries.get(id) || null;
  }

  values() {
    return Array.from(this.entries.values());
  }
}

/** Returns a tagged string for a CriticalUserModel. */
function transformCriticalUserModel(model) {
  if (!(model instanceof CriticalUserModel)) {
    return null;
  }
  return "[" + model.name + "]";
}

/** Creates a wrapped CriticalUserModel entry. */
function wrapCriticalUserModel(name, email) {
  const m = new CriticalUserModel(name, email);
  return { model: m, tag: "wrapped" };
}

module.exports = { CriticalUserModelCache, transformCriticalUserModel, wrapCriticalUserModel };
